### PR TITLE
Add First Iteration of Documentation of SDC Pipeline

### PR DESCRIPTION
**Pull request by dbarrous**
This PR includes the first iteration of the documentation of the SDC Pipeline architecture for review and feedback. It also adds the proper CI/CD tests for the automated doc building.

You can view the built docs here: https://sdc-aws-pipeline-architecture.readthedocs.io/en/add-docs/index.html

### CODE REVIEW

**Review by acrager — CHANGES_REQUESTED** [review]

File: docs/pipeline-diagrams.rst [hunk 1]
```diff
@@ -0,0 +1,18 @@
+.. _pipeline-diagrams:
+
+Pipeline Diagrams
+=================
+
+
+** Going to redo this as a normal image, because graphviz makes it look a bit too small and hard to read. **
```
[the remark this answers is not in the document]
Comment (dbarrous) [reply docs/pipeline-diagrams.rst]: Fixed in the most recent commit, ty!

File: docs/user-guide/aws-tag-overview.rst [hunk 2]
```diff
@@ -0,0 +1,17 @@
+.. _aws-tag-overview:
+
+AWS Tagging Overview
+====================
+AWS Tags are used to organize and categorize your AWS resources. You can use tags to manage access to your resources, to organize your resources, and to provide cost allocation information. For more information, see `Tagging AWS Resources <http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html>`__.
+
+How we use Tags
```
[the remark this answers is not in the document]
Comment (dbarrous) [reply docs/user-guide/aws-tag-overview.rst:7]: Fixed in the most recent commit, ty!

File: docs/user-guide/index.rst [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+User's Guide
+================
+
+Welcome to our User Guide. This guide will help you to get started with downloading this code and getting the pipeline set-up and running on your own AWS account. It will walk through each step of the process and provide you with the necessary information to get started.
+
+
+Prerequisites
+-------------
+* An AWS account
+* AWS Cli installed and configured
```
[the remark this answers is not in the document]
Comment (dbarrous) [reply docs/user-guide/index.rst]: Fixed in the most recent commit, ty!

**Review by acrager — APPROVED** [review]
Thanks! Diagrams render correctly, formatting in text looks great.

**Pull request merged**